src/validation.h

```diff
@@ -557,7 +557,14 @@ static const unsigned int REJECT_CONFLICT = 0x102;
 /** Dump the mempool to disk. */
 void DumpMempool();
 
+/** Result states for loading the mempool from disk */
+enum LoadMempoolResult {
```

Comment:
IMO a simple `true|false` would not work. You need to distinct between...
* not yet loaded = don't dump during shutdown
* user_aborted (shutdown) = don't dump during shutdown
* failed = dump during shutdown
* successful = dump during shutdown

Returning `false` in case of a user-triggered shutdown could work, but then you would need to return `true` in case of an input error (somehow not good).